doc: modem_lib: add how to use the uart trace backend on a custom board

eivindj-nordic opened the pull request:
Add description and code snippet on how to use the UART trace backend on a custom board.

Signed-off-by: Eivind Jølsgard <eivind.jolsgard@nordicsemi.no>

NordicBuilder commented:
### Integration test specification

| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |

Test spec set is empty, this is probabley due to no matching entry in [.github/test-spec.yml](https://github.com/nrfconnect/sdk-nrf/blob/main/.github/test-spec.yml)


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-nrf/job/master/26242/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-8527
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

divipillai reviewed (commented):
We should have a change log entry in the "documentation>added" section:

_A section about :ref:`modem_trace_backend_uart_custom_board` in the :ref:`nrf_modem_lib_readme` page._

eivindj-nordic commented:
> We should have a change log entry in the "documentation>added" section:
> 
> _A section about :ref:`modem_trace_backend_uart_custom_board` in the :ref:`nrf_modem_lib_readme` page._

@divipillai I have addressed your comments and added the changelog entry. Please have another look :)

lemrey reviewed (approved):
Thank you
